spec/controllers/annual_leave_requests_controller_spec.rb

```diff
@@ -78,11 +91,36 @@
       patch :update_status, params: {
         annual_leave_request_id: leave_request.id,
         annual_leave_request: {
+          status: "approved",
           confirm_approval: "confirmed",
         },
       }
 
       expect(notify_fake_client).to have_received(:send_email).with(approved_request_email_hash)
     end
+
+    it "sends an email to the line report when status is updated to 'denied'" do
+      patch :update_status, params: {
+        annual_leave_request_id: leave_request.id,
+        annual_leave_request: {
+          status: "denied",
+          denial_reason: "some valid reason",
+        },
+      }
+      leave_request.reload
```

Comment:
I feel a bit wary about sharing the `leave_request` object between these four tests, but I think in case its safe. Always a little bit prone to confusing mishaps though